### ISSUE TITLE
Spring Security redirection is not cleared in case the login did not happen

### ISSUE THREAD

**Issue by mrioan** [stormpath/stormpath-sdk-java#718]
### How to reproduce

1. Start `spring-security-webmvc`
2. Go to `http://localhost:8080/restricted`
3. You will see the login screen BUT DO NOT  login
4. Go to home
5. Go to `http://localhost:8080/login`
6. Login

#### Expected result
You are redirected to the `nextUri`

#### Actual outcome
You are redirected to `restricted`. 

### Issue
The `RequestCache` keeps the original intercepted large url in step 2 until the actual login ever happens

Will be tested via TCK: https://github.com/stormpath/stormpath-framework-tck/issues/274

**Comment by mrioan** [comment]
This issue is being tested by https://github.com/stormpath/stormpath-framework-tck/pull/288

**Comment by lhazlewood** [comment]
@mrioan how do I start the web server to UAT?  mvn tomcat7:run and mvn jetty:run do not work.

**Comment by mrioan** [comment]
@lhazlewood `mvn tomcat7:run` is the way I run it everyday. I just tried again in case I am confused but not problem at all, that is the proper way to run `examples/spring-security-webmvc`. What is the error that you are seeing?

BTW, this issue does not have a PR in the Java SDK. The issue was indirectly fixed by another unrelated PR. This issue was mainly kept alive in order to remember to review the TCK test that validates this issue is not happening again (I considered important to implement this test in the TCK for other SDKs as well). That TCK test is https://github.com/stormpath/stormpath-framework-tck/pull/288. I just checked and it has already been merged so this issue can be considered `done` as well.

Back to you @lhazlewood
